### multiline string

In [2]:
address = """ Eratous 12
    dapne
    17235
    Athens"""
print(address)

 Eratous 12
    dapne
    17235
    Athens


### In the following examples, the terms word, phrase, name, firstname and surname are all variable names.

In [6]:
phrase= "πέρα στους πέρα καμπους...  "

In [7]:
word = "olympics"
name = " Frederic"
firstname = "Paco"
surname = "Raban"

In [9]:
print(word.capitalize())

Olympics


In [10]:
name = firstname + surname

In [11]:
name

'PacoRaban'

In [15]:
text = " This is some text. "

In [16]:
print(text.strip(""))    ### first and last..."whatever"

 This is some text. 


In [17]:
print(word)

olympics


In [18]:
print(word.upper())

OLYMPICS


In [19]:
print(phrase.title())

Πέρα Στους Πέρα Καμπους...  


In [22]:
print("Hello world"[7:10])

orl


## 20


name = input("Your name ? ")
print(len(name))

### In Python, the terms "object method" and "object function" are often used interchangeably, but there is a subtle difference between the two.

An object method is a function that is associated with an object and is called on that object. For example, the upper method in the previous example (see CLASSS_INSTANCE_METHOD.ipynb) is a 
## string object method, and it's called on a string object. 
Object methods 
# operate on the data contained within an object and can modify that data OR return a new value based on the data.

# On the other hand, an object function is a standalone function that takes an object as an argument and operates on that object. For example,
## the len function in Python is a built-in object function, 

## which returns the number of items in an object. Object functions 

## do not modify the object passed as an argument and instead return a new value based on the object.

In general, the distinction between object methods and object functions is not critical and the terms are often used interchangeably. However, it's useful to understand the subtle difference between them to better understand the behavior and use of different functions and methods in Python.

## 21

In [10]:
firstname = input('First name ? ')
surname = input('surname ? ')
name = firstname+' '+surname
print(name.title())
print(len(name))

First name ?  kiki
surname ?  koko


Kiki Koko
9


In [11]:
nursery_rhyme = "savez vous planter les choux ?"
print(len(nursery_rhyme))
print(nursery_rhyme[3:9])

30
ez vou


## 26
## Pig Latin takes the first consonant of a word,
moves it to the end of the word and adds on an
“ay”. If a word begins with a vowel you just add
“way” to the end. For example, pig becomes igpay,
banana becomes ananabay, and aadvark becomes
aadvarkway. Create a program that will ask the
user to enter a word and change it into Pig Latin.
Make sure the new word is displayed in lower case.

In [29]:
given_word = input("give a word...")
word = given_word.lower()
length = len(word)
first = word[0]
rest = word[1:length]
if first !='a' and first !='e' and first !='i' and first !='o' and first !='u' and first !='y' :
   newword = rest+first+'ay'
else : newword = word+'way'     
print(newword)    

give a word... Abraham


abrahamway


In [32]:
given_word = input("give a word...")
word = given_word.lower()
length = len(word)
first = word[0]
rest = word[1:length]
if first =='a' or first =='e' or first =='i' or first =='o' or first =='u' or first =='y' :
   newword = word+'way'
else :  newword = rest+first+'ay'    
print(newword)    

give a word... koki


okikay


In [36]:
given_word = input("give a word...")
word = given_word.lower()
length = len(word)
first = word[0]
rest = word[1:length]
vowel = ['a' , 'e' , 'i' , 'o' , 'u' , 'y']
if first in vowel :
   newword = word+'way'
else :  newword = rest+first+'ay'    
print(newword)  

give a word... liko


ikolay


In [1]:
import math
math.pi


3.141592653589793


# f"      "
# f" blabla {}"
# f" comment_string {item_to_format : format}"  

### The statement print(f" pi = {pi:.2f}") in Python is using 

# f-strings (formatted string literals).

### The f before the string indicates that this is an f-string and the expressions inside the curly braces {} are evaluated and replaced with their values.
### The expression inside the braces {pi:.2f} specifies the value of the variable pi (which should be a float value) and the formatting that should be applied to it.
### The .2f part specifies the format of the float value. The .2 specifies that only 2 digits should be after the decimal point, and the f specifies that the value should be formatted as a float.
### So, when the statement is executed, the value of pi is inserted into the string in the place of {pi:.2f} and the resulting string is printed to the console.

For example, if pi = 3.14159265359, then the statement print(f" pi = {pi:.2f}") will print the following output to the console:

In [7]:
print(f"{pi:.2f}")           # f-string

3.14


In [8]:
print(f" pi = {.2f}")  

SyntaxError: f-string: invalid syntax (2731548582.py, line 1)

In [18]:
print(f" pi = {pi.2f}")  

SyntaxError: f-string: invalid syntax (1149503982.py, line 1)

In [10]:
print(f" Greek π = {pi:.2f}")  

 Greek π = 3.14


In [13]:
x = math.sqrt(2)
print(x)

1.4142135623730951


In [15]:
print(round(x , 7))

1.4142136


In [17]:
print(f"x = {x:.4f}")

x = 1.4142


In [20]:
R = float(input("Give the radius of the circle..."))
print(f"The area is... {pi*R**2 : .3f}")

Give the radius of the circle... 5


The area is...  78.540


# f"" with multiple {}

# f"foo {} bar {} bart {} ...."

In [24]:
R = float(input("Give the radius of the circle..."))
print(f"The area of a circle of Radius {R : .2f}  is  {pi*R**2 : .3f}")

Give the radius of the circle... 5


The area of a circle of Radius  5.00  is   78.540


In [6]:
dvdnd = int(input("dividend is   .... "))
dvdr = int(input("divider is ..... "))
print(f"quotient is {dvdnd//dvdr : .0f}")
print(f"remainder is {dvdnd % dvdr : .0f}")

dividend is   ....  7
divider is .....  3


quotient is  2
remainder is  1


In [10]:
print("1) rectangle")
print("2) triangle")
scheme = int(input("select scheme.... "))
if scheme == 1 :
    a = float(input(" plevra... "))
    print (f"rectangle area = {a**2 : .2f}")
elif scheme ==2 :
    b = float(input("triangle base..."))
    h = float(input("triangle height... "))
    print(f" the area of the triangle is...  {b*h/2 : .2f}")
else :
    print("invalid selection")
    

1) rectangle
2) triangle


select scheme....  5


invalid selection
